collector/lib/NetworkStatusInspector.h

```diff
@@ -0,0 +1,38 @@
+#ifndef COLLECTOR_NETWORKSTATUSINSPECTOR_H
+#define COLLECTOR_NETWORKSTATUSINSPECTOR_H
+
+#include <memory>
+#include <optional>
+#include <unordered_map>
+
+#include <json/writer.h>
+
+#include "IntrospectionEndpoint.h"
+
+namespace collector {
+
+class ConnectionTracker;
+
+class NetworkStatusInspector : public IntrospectionEndpoint {
+ public:
+  static const std::string kBaseRoute;
+  static const std::string kEndpointRoute;
+  static const std::string kConnectionRoute;
+
+  NetworkStatusInspector(const std::shared_ptr<ConnectionTracker> conntracker);
+
+  // implementation of CivetHandler
+  bool handleGet(CivetServer* server, struct mg_connection* conn) override;
+
+ private:
+  const std::shared_ptr<ConnectionTracker> conntracker_;
+  Json::StreamWriterBuilder jsonStreamWriterBuilder_;
```

Comment:
Would it make sense to turn this into a `Json::StreamWriter` and use that for printing? Right now we are creating and destroying one on every API call, or is a race condition on multiple calls a concern and I'm just being dumb?

Reply:
The FastWriter seems just right for that purpose, I replaced with a single instance of it.